docs/optlib.rst

```diff
@@ -181,269 +206,413 @@ Tips about writing option file
 
-`{scope=set}`
+``{scope=set}``
 
 	Clear then push.
 
-`{placeholder}`
+``{placeholder}``
 
-	Don't print a tag captured with a regex pattern
-	to a tag file.
-	This is useful when you need to push non-named context
-	information to the stack.  Well known non-named scope in C
-	language is established with `{`. non-named scope is never
-	appeared in tags file as name or scope name.  However, pushing
-	it is important to balance `push` and `pop`.
+	Don't print a tag captured with a regex pattern to a tag file. This is
+	useful when you need to push non-named context information to the stack.
+	Well known non-named scope in C language is established with `{`. A non-
+	named scope never appears in tags file as a name or scope name.  However,
+	pushing it is important to balance ``push`` and ``pop``.
 
-Example 1::
+Example 1:
 
-	$ cat /tmp/input.foo
+.. code-block:: python
+
+	# in /tmp/input.foo
 	class foo:
 	def bar(baz):
 		print(baz)
 	class goo:
 	def gar(gaz):
 		print(gaz)
 
-	$ cat /tmp/foo.ctags
-	--langdef=foo
-		--map-foo=+.foo
-		--regex-foo=/^class[[:blank:]]+([[:alpha:]]+):/\1/c,class/{scope=set}
-		--regex-foo=/^[[:blank:]]+def[[:blank:]]+([[:alpha:]]+).*:/\1/d,definition/{scope=ref}
+.. code-block:: perl
+
+	# in /tmp/foo.ctags:
+	--langdef=Foo
+	--map-Foo=+.foo
+	
+	--regex-Foo=/^class[[:blank:]]+([[:alpha:]]+):/\1/c,class/{scope=set}
+	--regex-Foo=/^[[:blank:]]+def[[:blank:]]+([[:alpha:]]+).*:/\1/d,definition/{scope=ref}
 
-	$ ~/var/ctags/ctags --options=/tmp/foo.ctags -o - /tmp/input.foo
+.. code-block:: console
+
+	$ ctags --options=/tmp/foo.ctags -o - /tmp/input.foo
 	bar	/tmp/input.foo	/^    def bar(baz):$/;"	d	class:foo
 	foo	/tmp/input.foo	/^class foo:$/;"	c
 	gar	/tmp/input.foo	/^    def gar(gaz):$/;"	d	class:goo
 	goo	/tmp/input.foo	/^class goo:$/;"	c
 
 
-Example 2::
+Example 2:
+
+.. code-block:: c
 
-	$ cat /tmp/input.pp
+	// in /tmp/input.pp
 	class foo {
-	include bar
+		int bar;
 	}
 
-	$ cat /tmp/pp.ctags
+.. code-block:: perl
+
+	# in /tmp/pp.ctags:
 	--langdef=pp
-		--map-pp=+.pp
-		--regex-pp=/^class[[:blank:]]*([[:alnum:]]+)[[[:blank:]]]*\{/\1/c,class,classes/{scope=push}
-		--regex-pp=/^[[:blank:]]*include[[:blank:]]*([[:alnum:]]+).*/\1/i,include,includes/{scope=ref}
-		--regex-pp=/^[[:blank:]]*\}.*//{scope=pop}{exclusive}
+	--map-pp=+.pp
 
-	$ ~/var/ctags/ctags --options=/tmp/pp.ctags -o - /tmp/input.pp
-	bar	/tmp/input.pp	/^    include bar$/;"	i	class:foo
-	foo	/tmp/input.pp	/^class foo {$/;"	c
+	--regex-pp=/^[[:blank:]]*\}//{scope=pop}{exclusive}
+	--regex-pp=/^class[[:blank:]]*([[:alnum:]]+)[[[:blank:]]]*\{/\1/c,class,classes/{scope=push}
+	--regex-pp=/^[[:blank:]]*int[[:blank:]]*([[:alnum:]]+)/\1/v,variable,variables/{scope=ref}
 
+.. code-block:: console
+
+	$ ctags --options=/tmp/pp.ctags -o - /tmp/input.pp
+	bar	/tmp/input.pp	/^    include bar$/;"	v	class:foo
+	foo	/tmp/input.pp	/^class foo {$/;"	c
 
-NOTE: Giving a scope long flag implies setting `useCork` of the parser
-to `TRUE`. See `cork API`.
```

Comment:
Could keep the above sentence about cork ?
I think the sentence helps a developer understand u-ctags internal.

Reply:
But an optlib developer (i.e., a person who creates optlib files) does not need to know anything about it, and will be confused why they should care. I mean it's just an internal implementation detail, like setting the `parserDefinition.method` flag `METHOD_REGEX`. No?

This is an example of something that confused me when reading the doc the first time: who's the target audience? Is it general users of ctags and authors of optlib files, or is it for C-code developers?

Reply:
Though I have not written well as I expect, this document is for rather C-code developers.

I assume optlib developers (== general users) read ctags-optlib.7 man page.
I will write more about this topic in #1835.